tests/backup/backup_upgrade_test.go

```diff
@@ -1546,3 +1547,371 @@ var _ = Describe("{PXBackupClusterUpgradeTest}", Label(TestCaseLabelsMap[PXBacku
+				azureCredConfig = cloudCred.CloudCredential.CloudCredentialInfo.GetAzureConfig()
+				log.Infof("Cred details after update: %v", azureCredConfig)
+
+				log.InfoD("Restoring the backed up namespaces from backup taken before and after upgrade")
+				restoreName := fmt.Sprintf("%s-%s", "restore-pre-upgrade-backup", RandomString(4))
+				namespaceMap = make(map[string]string)
+				for _, appCtx := range scheduledAppContexts {
+					namespaceMap[appCtx.ScheduleOptions.Namespace] = appCtx.ScheduleOptions.Namespace + "-res-old"
+				}
+				log.InfoD("Restoring from the [%s] backup", preUpgradeBackupName)
+				err = os.Setenv("BACKUP_TYPE", "azure")
+				log.FailOnError(err, "Setting BACKUP_TYPE env variable to azure")
+				err = CreateRestoreWithValidation(ctx, restoreName, preUpgradeBackupName, namespaceMap, make(map[string]string), DestinationClusterName, BackupOrgID, scheduledAppContexts)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of restore [%s]", restoreName))
+
+				restoreName = fmt.Sprintf("%s-%s", "restore-post-upgrade-backup", RandomString(4))
+				namespaceMap = make(map[string]string)
+				for _, appCtx := range scheduledAppContexts {
+					namespaceMap[appCtx.ScheduleOptions.Namespace] = appCtx.ScheduleOptions.Namespace + "-res-new"
+				}
+				log.InfoD("Restoring from the [%s] backup", postUpgradeBackupName)
+				err = os.Setenv("BACKUP_TYPE", "native_csi")
+				log.FailOnError(err, "Setting BACKUP_TYPE env variable to native_csi")
+				err = CreateRestoreWithValidation(ctx, restoreName, postUpgradeBackupName, namespaceMap, make(map[string]string), DestinationClusterName, BackupOrgID, scheduledAppContexts)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of restore [%s]", restoreName))
+
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(make([]*scheduler.Context, 0))
+		defer func() {
+			log.Infof("Unsetting BACKUP_TYPE env variable")
+			err := os.Unsetenv("BACKUP_TYPE")
+			log.FailOnError(err, "Unsetting BACKUP_TYPE env variable")
+		}()
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		log.Info("Destroying scheduled apps on source cluster")
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		err = DestroyAppsWithData(scheduledAppContexts, opts, controlChannel, errorGroup)
+		log.FailOnError(err, "Data validations failed")
+		// Need to delete the cluster before deleting the cloud credential
+		clusterNames := []string{SourceClusterName, DestinationClusterName}
+		for _, clusterName := range clusterNames {
+			log.InfoD("Deleting cluster [%s]", clusterName)
+			err = DeleteCluster(clusterName, BackupOrgID, ctx, true)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster [%s]", clusterName))
+		}
+		// Enumerate the clusters and verify that the clusters are deleted
+		t := func() (interface{}, bool, error) {
```

Comment:
better to make use of WaitForClusterDeletion here.

Reply:
Done